lightning/src/ln/onion_utils.rs

```diff
@@ -1773,9 +1773,10 @@ impl From<&HTLCFailReason> for HTLCHandlingFailureReason {
 	}
 }
 
+/// Contains the information required to construct the failure message for a failed HTLC.
 #[derive(Clone)] // See Channel::revoke_and_ack for why, tl;dr: Rust bug
 #[cfg_attr(test, derive(PartialEq))]
-pub(super) struct HTLCFailReason(HTLCFailReasonRepr);
+pub(super) struct HTLCFailurePayload(HTLCFailReasonRepr);
```

Comment:
Embedded inside there is still a reason? Payload with a reason repr 🤔

Reply:
Rename the internal struct as well?

Open to other naming suggestions here, aim is to differentiate this from the `HTLCHandlingFailureReason` a bit.

Reply:
I wanted to suggest renaming the internal struct too indeed. But mabe I am not fully convinced that the rename is enough of an improvement. Reason and payload - it might also create confusion because now they seem different things. FullReason could be another option. Or drop the rename if it isn't a clear improvement.

Reply:
@valentinewallace originally suggested [RawOnionFailure](https://github.com/lightningdevkit/rust-lightning/pull/3700#discussion_r2029452317), perhaps `RawOnionFailure`/`WireOnionFailure`/`InternalOnionFailure` fit better here.

I don't feel very strongly about this, so happy to drop or pick another name.

Reply:
Hm none of them resonate convincingly with me. Fine with any decision here.